app/modules/policy_handler.rb

```diff
@@ -0,0 +1,9 @@
+module PolicyHandler
```

Comment:
your module don't know about method `policy(menu)`.

That's duck typing, yeah, 
but IMO would be better to move method `policy` here below `protected` access modifier